uefi/src/proto/shell_params.rs

```diff
@@ -0,0 +1,45 @@
+//! `ShellParams` protocol
+
+use crate::proto::unsafe_protocol;
+use crate::{data_types, Char16};
+use core::slice::from_raw_parts;
+use uefi_raw::protocol::shell_params::ShellParametersProtocol;
+
+#[cfg(feature = "alloc")]
+use crate::CStr16;
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+#[cfg(feature = "alloc")]
+use alloc::string::ToString;
+
+/// The ShellParameters protocol.
+#[derive(Debug)]
+#[repr(transparent)]
+#[unsafe_protocol(ShellParametersProtocol::GUID)]
+pub struct ShellParameters(ShellParametersProtocol);
+
+impl ShellParameters {
+    /// Get an iterator of the shell parameter arguments
+    #[cfg(feature = "alloc")]
+    pub fn get_args(&self) -> impl Iterator<Item = String> {
```

Comment:
naming nit: drop the `get_` prefix on these methods (https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter)